libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/NavigationMapRoute.java

```diff
@@ -445,7 +445,10 @@ private void addDirectionWaypoints() {
   }
 
   private void addUpcomingManeuverArrow(RouteProgress routeProgress) {
-    if (routeProgress.upcomingStepPoints() == null || routeProgress.upcomingStepPoints().size() < TWO_MANEUVERS) {
+    final boolean invalidUpcomingStepPoints = routeProgress.upcomingStepPoints() == null
```

Comment:
Why does `invalidUpcomingStepPoints` need to be `final`?

Reply:
I prefer to have variables immutable, if they don't have to be mutable. I can remove the `final` if you don't like it? I don't have a strong opinion on this.

Reply:
Yeah, fair point. In the case of local variables, I personally prefer to avoid this. It causes visual clutter, and is generally unnecessary - a function should be short enough (preferably only doing one clear thing) that lets you quickly see that you are modifying something that you shouldn't. But yeah I agree this is a matter of taste 😅

For consistency with the rest of the code in which we're not including the `final` modifier for local variables, I'd say that removing it is the best solution to take here. What do you think?

Reply:
Not a problem, I'll remove the final statement. Code should be consistent :+1:.